layers.go

```diff
@@ -318,9 +321,11 @@ func (r *layerStore) Load() error {
 			ids[layer.ID] = layers[n]
 			idlist = append(idlist, layer.ID)
 			for _, name := range layer.Names {
-				if conflict, ok := names[name]; ok {
-					r.removeName(conflict, name)
-					shouldSave = true
+				if r.Locked() {
```

Comment:
Please add a comment why this only applies when the store is locked.

Reply:
Probably not needed, I will give a removal of this part another try.

Reply:
Looks like it is working, I'm now vendoring it into the related PRs.